core/src/main/java/com/graphhopper/storage/GraphHopperStorage.java

```diff
@@ -270,10 +268,14 @@ public boolean loadExisting() {
 
             checkIfConfiguredAndLoadedWeightingsCompatible();
 
-            chEntries.forEach(cg -> {
-                if (!cg.chStore.loadExisting())
-                    throw new IllegalStateException("Cannot load " + cg);
-            });
+            List<Callable<String>> callables = chEntries.stream().map(c -> (Callable<String>) () -> {
+                        if (!c.chStore.loadExisting())
+                            throw new IllegalStateException("Cannot load " + c);
+                        return c.chConfig.getName();
+                    })
+                    .collect(Collectors.toList());
+            int numThreads = Math.max(1, Math.min(4, callables.size()));
```

Comment:
We used to only use one thread here. The `prepare.ch.threads` parameter is not available here, so I hard-coded the default to 4 (max). Should be ok, because the memory usage won't be increased because of this.

Reply:
If memory consumption doesn't scale with the threadcount here, why don't we use all available cores?

```java
int numThreads = Math.max(1, Math.min(Runtime.availableProcessors(), callables.size()));
```

Reply:
Yes, this would also be possible, I guess. I was just trying to be conservative. If we did that we could also just use `.parallelStream()`.

Reply:
Using `parallelStream()` has some sideeffects as it's ForkJoinPool is shared by the whole JVM. Probably not that bad for standalone GraphHopper but a pitfall people which integrate it as a library.

Reply:
Ok, good to know. :+1: